tests/multi_ptr/multi_ptr_subscript_op_common.h

```diff
@@ -0,0 +1,166 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+#ifndef __SYCLCTS_TESTS_MULTI_PTR_SUBSCRIPT_OP_H
+#define __SYCLCTS_TESTS_MULTI_PTR_SUBSCRIPT_OP_H
+
+#include "multi_ptr_common.h"
+
+namespace multi_ptr_subscript_op {
+
+/**
+ * @brief Provides functions for verification multi_ptr subscript operator that
+ *        it returns correct data
+ * @tparam T Current data type
+ * @tparam AddrSpaceT sycl::access::address_space enumeration's field
+ * @tparam IsDecoratedT sycl::access::decorated enumeration's field
+ */
+template <typename T, typename AddrSpaceT, typename IsDecoratedT>
+class run_subscript_op_tests {
```

Comment:
It is unclear to me why this is not just a function instead of class + `operator()`.

Reply:
Because I call `for_all_combinations` to invoke it, this function takes functor to run it. If there is no need to test subscript operator in all address spaces I can refactor it.